MonstieWash/Assets/Scripts/Animation Control/MonsterAnimControl.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[Serializable]
+public class MoodToAnimName
+{
+    public string moodName;
+    public string animName;
+}
+
+public class MonsterAnimControl : MonoBehaviour
+{
+    private MonsterBrain m_mimicAI;   
+    private string m_currentMood;   // tracks the monster's current mood (accurate to last frame)
+    private Animator m_myAnimator;
+
+    [SerializeField] private List<MoodToAnimName> moodToAnimationMap; // maps the name of moods to their animation names
+
+    void Start()
+    {
+        m_myAnimator = GetComponent<Animator>();
+    }
+
+    private void OnEnable()
+    {
+        if (m_mimicAI == null)
+        {
+            m_mimicAI = FindFirstObjectByType<MonsterBrain>();
+        }
+    }
```

Comment:
I'm not sure why this is set up in `OnEnable` like this but it makes much more sense for it to be done in `Start` so you don't need to check if it's already been set. Also don't forget to add the private modifier to the method.
```suggestion
    private void Start()
    {
        m_myAnimator = GetComponent<Animator>();
        m_mimicAI = FindFirstObjectByType<MonsterBrain>();
    }
```